pydantic_settings/sources/base.py

```diff
@@ -353,7 +355,10 @@ def _extract_field_info(self, field: FieldInfo, field_name: str) -> list[tuple[s
                 field_info.append((v_alias, self._apply_case_sensitive(v_alias), False))
 
         if not v_alias or self.config.get('populate_by_name', False):
-            if is_union_origin(get_origin(field.annotation)) and _union_is_complex(field.annotation, field.metadata):
+            annotation = field.annotation
+            if annotation and typing_objects.is_typealiastype(annotation):
```

Comment:
Is there a specific need to check for the truthiness of `annotation` first? (Which should always evaluate to `True` unless something unexpected is happening).

Reply:
@Viicos you are absolutely right, I hadn't took into consideration the generic type alias case. As you will see, the `annotation` truthiness check is back as mypy complains that annotation might be `None`.